app/celery/process_ses_receipts_tasks.py

```diff
@@ -36,31 +35,52 @@
 
 @notify_celery.task(bind=True, name='process-ses-result', max_retries=5, default_retry_delay=300)
 @statsd(namespace='tasks')
-def process_ses_results(  # noqa: C901 (too complex 14 > 10)
+def process_ses_results(  # noqa: C901 (too complex 19 > 10)
     self,
     response,
 ):
     current_app.logger.debug('Full SES result response: %s', response)
     try:
         ses_message = json.loads(response['Message'])
+
+        reference = ses_message['mail']['messageId']
+        if not reference:
+            current_app.logger.warning(
+                'SES complaint: unable to lookup notification, messageId (reference) was None | ses_message: %s',
+                ses_message,
+            )
+            return
+
         notification_type = ses_message.get('eventType')
 
         if notification_type == 'Bounce':
             # Bounces have ran their course with AWS and should be considered done. Clients can retry soft bounces.
             notification_type = determine_notification_bounce_type(notification_type, ses_message)
         elif notification_type == 'Complaint':
-            complaint, notification, recipient_email = handle_ses_complaint(ses_message)
-            return publish_complaint(complaint, notification, recipient_email)
+            try:
+                notification = notifications_dao.dao_get_notification_history_by_reference(reference)
+            except Exception:
+                # we expect results or no results but it could be multiple results
+                message_time = iso8601.parse_date(ses_message['mail']['timestamp']).replace(tzinfo=None)
+                if datetime.utcnow() - message_time < timedelta(minutes=5):
+                    self.retry(queue=QueueNames.RETRY)
+                else:
+                    current_app.logger.warning('SES complaint: notification not found | reference: %s', reference)
+                return
+
+            complaint, recipient_email = handle_ses_complaint(ses_message, notification)
+            publish_complaint(complaint, notification, recipient_email)
+            return
 
```

Comment:
You need an "else" here because notification_type could be None.  Maybe that shouldn't happen, but it's possible according to the code.  Another option would be to `assert notification_type is not None`.  I like to use assertions to codify assumptions.

Yet another option would be to test above `if notification_type is None` and early-out accordingly.